S32-num/modf.t

```diff
@@ -0,0 +1,91 @@
+use Test;
+
+# L<S32::Numeric/Numeric/"=item modf">
+
+plan 240;
+
+my %n =
+    '-123457e-3' => [-123, -0.457,            3],
+    '-10'        => [-10,   0,                0],
+    '-0x10'      => [-16,   0,                0],
+    '-0o10'      => [-8,    0,                0],
+    '-0xa'       => [-10,   0,                0],
+    '-0o127'     => [-87,   0,                0],
+    '-100'       => [-100,  0,                0],
+    '-5.9'       => [-5,   -0.9,              1],
+    '-5.499'     => [-5,   -0.499,            3],
+    '-2'         => [-2,    0,                0],
+    '-3/2'       => [-1,   -0.5,              1],
+    '-1.5e0'     => [-1,   -0.5,              1],
+    '-1.4999'    => [-1,   -0.4999,           4],
+    '-1.23456'   => [-1,   -0.23456,          5],
+    '-1'         => [-1,    0,                0],
+    '-0.5'       => [ 0,   -0.5,              1],
+    '-0.499'     => [ 0,   -0.499,            3],
+    '-0.1'       => [ 0,   -0.1,              1],
+    '-0'         => [ 0,    0,                0],
+    '-1.5e-10'   => [ 0,   -0.000_000_000_15, 11], # -0.000_000_000_15
+;
+
+
+# test the routine
+for %n.kv -> $val is copy, $arr {
```

Comment:
Just an advise. You can shorten the code here by de-structuring the `$arr` in place:

```raku
for %n.kv -> $val is copy, ($int-part, $frac-part, $places) {
```

Aside of this, unless you want to intentionally make the test random-ordered, I consider use of hash redundant here. To my view, 

```raku
my @n = [ '-5.9', -5, -0.9, 1],
        ['-123457e-3', -123, -0.457, 3],
...

for @n -> ($val is copy, $int-part, $frac-part, $places) {
```

makes the code even more readable.

Reply:
@vrurg, I changed this to draft earlier because I'm still working on the Rakudo PR as well as the roast test code. I appreciate the comments, though, and you will probably see them used in the final results.

Reply:
Ok, will try to remember keeping my eye on this. Anyway, thank you for PRs!

Reply:
You're very welcome, and I 'm learning a lot from this "exercise!"